callable/src/main/java/io/grpc/callable/Callable.java

```diff
@@ -0,0 +1,423 @@
+   * </pre>
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(MethodDescriptor<RequestT, ResponseT> descriptor) {
+    return create(CallableDescriptor.create(descriptor));
+  }
+
+  /**
+   * Returns a callable which executes the method described by a {@link CallableDescriptor}.
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(final CallableDescriptor<RequestT, ResponseT> descriptor) {
+    return new Callable<RequestT, ResponseT>() {
+      @Override public ClientCall<RequestT, ResponseT> newCall(Channel channel) {
+        if (channel == null) {
+          throw new IllegalStateException(String.format(
+              "unbound callable for method '%s' requires a channel for execution",
+              descriptor.getMethodDescriptor().getFullMethodName()));
+        }
+        return channel.newCall(descriptor.getMethodDescriptor(), CallOptions.DEFAULT);
+      }
+
+      @Override public CallableDescriptor<RequestT, ResponseT> getDescriptor() {
+        return descriptor;
+      }
+
+      @Override public String toString() {
+        return descriptor.getMethodDescriptor().getFullMethodName();
+      }
+    };
+  }
+
+  /**
+   * Returns a callable which executes the given function asynchronously on each provided
+   * input. The supplied executor is used for creating tasks for each input. Example:
+   *
+   * <pre>
+   *  Callable.Transformer&lt;RequestT, ResponseT> transformer = ...;
+   *  Response response = Callable.create(transformer, executor).call(channel, request);
+   * </pre>
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(Transformer<? super RequestT, ? extends ResponseT> transformer, Executor executor) {
+    return new TransformingCallable<RequestT, ResponseT>(transformer, executor);
+  }
+
+
+  /**
+   * Returns a callable which executes the given function immediately on each provided input.
+   * Similar as {@link #create(Transformer, Executor)} but does not operate asynchronously and does
+   * not require an executor.
+   *
+   * <p>Note that the callable returned by this method does not respect flow control. Some
+   * operations applied to it may deadlock because of this. However, it is safe to use this
+   * callable in the context of a {@link #followedBy(Callable)} operation, which is the major
+   * use cases for transformers. But if you use a transformer to simulate a real rpc
+   * you should use {@link #create(Transformer, Executor)} instead.
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(Transformer<? super RequestT, ? extends ResponseT> transformer) {
+    return new TransformingCallable<RequestT, ResponseT>(transformer, null);
+  }
+
+  /**
+   * Interface for a transformer. It can throw a {@link StatusException} to indicate an error.
+   */
+  public interface Transformer<RequestT, ResponseT> {
```

Comment:
This doesn't seem right. You're taking a requestT and returning a responseT which if the naming you've used here is correct would imply that you're going to use this generate responses from requests (i.e not make an API call at all).

I'm assuming what you really want to do is to transform RequestT into RequestTPrime so that this can be used to apply a transformation to one side of the duplex message stream. I'd also assume that what we really want is to be able to synthesize something like

Callable<Integer /** book id */, Book> from Callable<GetBookRequest, GetBookResponse> to simply the interface we can present to users.